src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/RabbitMessageGroupBatchRouter.kt

```diff
@@ -39,77 +43,76 @@ class RabbitMessageGroupBatchRouter : AbstractRabbitRouter<MessageGroupBatch>()
 
     override fun splitAndFilter(
         message: MessageGroupBatch,
-        pinConfiguration: @NotNull QueueConfiguration
+        pinConfiguration: @NotNull QueueConfiguration,
+        pinName: PinName
     ): @NotNull Collection<MessageGroupBatch> {
         val builders = mutableListOf<Builder>()
-
         getMessages(message).forEach { group ->
-            val originalMessages = group.messagesList
-            val parsedRawPartition = originalMessages.partition { it.hasMessage() }
-
-            if (parsedRawPartition.first.size == originalMessages.size || parsedRawPartition.second.size == originalMessages.size) {
-                val forFilter = if (parsedRawPartition.first.isEmpty()) parsedRawPartition.second else parsedRawPartition.first
-                forFilter.forEach {
-                    if (filter(it, pinConfiguration)) {
-                        builders.add(createBatchBuilder().addGroups(group))
-                    }
-                }
+            val parsedToRawPair = group.messagesList.partition { it.hasMessage() }
```

Comment:
Optimize code to reduce creation of intermediate objects

Reply:
done